src/lib/mdns/Advertiser.h

```diff
@@ -51,11 +57,34 @@ class BaseAdvertisingParams
         return *reinterpret_cast<Derived *>(this);
     }
     bool IsIPv4Enabled() const { return mEnableIPv4; }
+    Derived & SetMac(chip::ByteSpan mac)
+    {
+        mMac = chip::ByteSpan(mMacStorage, std::min(mac.size(), kMaxMacSize));
+        memcpy(mMacStorage, mac.data(), mMac.size());
+        // Fill in the host name here so it's available.
+        size_t idx = 0;
+        for (size_t i = 0; i < mMac.size(); ++i)
+        {
+            if (idx > kMaxHostnameSize - 2)
+            {
+                ChipLogError(Discovery, "Failed to allocate full hostname");
+                break;
+            }
+            idx += snprintf(mHostname + idx, 3, "%02X", mac.data()[i]);
+        }
+        return *reinterpret_cast<Derived *>(this);
+    }
+    const chip::ByteSpan GetMac() const { return mMac; }
+    const char * GetHostname() const { return mHostname; }
 
 private:
-    uint16_t mPort   = CHIP_PORT;
-    bool mEnableIPv4 = true;
-};
+    uint16_t mPort                   = CHIP_PORT;
+    bool mEnableIPv4                 = true;
+    uint8_t mMacStorage[kMaxMacSize] = {};
+    chip::ByteSpan mMac              = chip::ByteSpan(mMacStorage, kMaxMacSize);
+    // Add 1 for the end nullptr.
+    char mHostname[kMaxHostnameSize + 1] = "000000000000";
```

Comment:
Why is it "000000000000"? The array is 17 bytes size and there are 12 characters in the string. Wouldn't be enough just to set `/0`, as a first character to mark array empty?

Reply:
wifi mac is 6 bytes, which translates to 12 ascii chars. I figured all zeros would be a good way to indicate that something was not correctly initialized. I'm not sure what a blank hostname will do to the mdns clients - you'll end up with a name that's just .local, which could potentially cause issues. 

I can add a comment to that effect. If you have a strong preference for empty, I'm fine to go that route too.

Reply:
Honestly I don't know how the empty host name will be treated by the mDNS, but I guess that is doesn't have much sense, so maybe when hostname could not be assigned we should assert? For example SRP doesn't really care about what host name contains, so assigning all zeros will result in creating record in SRP server for host: 000000000000.local. Imagine that for some reason assigning host name will fail for multiple devices and all of them will try to register as the same host called 000000000000.

Reply:
Code has changed - this is now removed.